compiler/fir/java/src/org/jetbrains/kotlin/fir/java/deserialization/JvmBinaryAnnotationDeserializer.kt

```diff
@@ -42,36 +45,69 @@ class JvmBinaryAnnotationDeserializer(
+        typeTable: TypeTable
+    ): List<FirAnnotationCall> {
+        val signature = getCallableSignature(functionProto, nameResolver, typeTable, CallableKind.FUNCTION) ?: return emptyList()
+        return findJvmBinaryClassAndLoadMemberAnnotations(containerSource, signature)
+    }
+
     override fun loadPropertyGetterAnnotations(
         containerSource: DeserializedContainerSource?,
         propertyProto: ProtoBuf.Property,
         nameResolver: NameResolver,
+        typeTable: TypeTable,
         getterFlags: Int
     ): List<FirAnnotationCall> {
-        val signature = getCallableSignature(propertyProto, nameResolver, CallableKind.PROPERTY_GETTER) ?: return emptyList()
-        val kotlinClass = containerSource?.toKotlinJvmBinaryClass() ?: return emptyList()
-        return loadMemberAnnotations(kotlinClass).memberAnnotations[signature] ?: emptyList()
+        val signature = getCallableSignature(propertyProto, nameResolver, typeTable, CallableKind.PROPERTY_GETTER) ?: return emptyList()
+        return findJvmBinaryClassAndLoadMemberAnnotations(containerSource, signature)
     }
 
     override fun loadPropertySetterAnnotations(
         containerSource: DeserializedContainerSource?,
         propertyProto: ProtoBuf.Property,
         nameResolver: NameResolver,
+        typeTable: TypeTable,
         setterFlags: Int
     ): List<FirAnnotationCall> {
-        val signature = getCallableSignature(propertyProto, nameResolver, CallableKind.PROPERTY_SETTER) ?: return emptyList()
-        val kotlinClass = containerSource?.toKotlinJvmBinaryClass() ?: return emptyList()
-        return loadMemberAnnotations(kotlinClass).memberAnnotations[signature] ?: emptyList()
+        val signature = getCallableSignature(propertyProto, nameResolver, typeTable, CallableKind.PROPERTY_SETTER) ?: return emptyList()
+        return findJvmBinaryClassAndLoadMemberAnnotations(containerSource, signature)
     }
 
     private fun getCallableSignature(
         proto: MessageLite,
         nameResolver: NameResolver,
+        typeTable: TypeTable,
         kind: CallableKind
     ): MemberSignature? {
         return when (proto) {
-            // TODO: ProtoBuf.Constructor
-            // TODO: ProtoBuf.Function
+            is ProtoBuf.Constructor -> {
+                MemberSignature.fromJvmMemberSignature(
+                    JvmProtoBufUtil.getJvmConstructorSignature(proto, nameResolver, typeTable) ?: return null
+                )
+            }
+            is ProtoBuf.Function -> {
+                val signature = JvmProtoBufUtil.getJvmMethodSignature(proto, nameResolver, typeTable) ?: return null
+                // TODO: Investigate why annotations for accessors affect resolution, resulting in dangling type parameter.
+                //   regressions: Fir2IrTextTest.Declarations.test*LevelProperties
+                if (signature.name.startsWith("get") || signature.name.startsWith("set")) {
+                    return null
+                }
```

Comment:
To be honest, all works have done a day ago, but it took me a while to deal with those regressions. :( Without these restriction, some with ambiguous resolutions are resolved (and unmuted, more clear IR texts, etc.), but some are simply failing with remaining type parameter at the conversion phase. I'll keep investigating it, but any hints, advices, insights, etc. would be appreciated, @mglukhikh